cmd/kubeadm/app/cmd/join.go

```diff
@@ -145,6 +145,11 @@ func (j *Join) Run(out io.Writer) error {
 		return err
 	}
 
+	err = kubenode.CheckForNodeNameDuplicates(connectionDetails)
```

Comment:
Have you manually verified this works?
Just wondering if it should be before or after the TLS bootstrap...